src/lib/activity/evm/parse/gr-v3.ts

```diff
@@ -0,0 +1,205 @@
+    blockExplorerUrl: item.explorers?.at(0)?.url,
+    operations,
+    operationsCount: gasOperation ? logEvents.length + 1 : logEvents.length,
+    addedAt,
+    status: item.successful ? ActivityStatus.applied : ActivityStatus.failed
+  };
+}
+
+function parseLogEvent(
+  logEvent: LogEvent,
+  item: Transaction,
+  /** Lower-cased */
+  accountAddress: string
+): EvmOperation {
+  const contractAddress = logEvent.sender_address ?? undefined;
+
+  if (!logEvent.decoded?.params) return { kind: ActivityOperKindEnum.interaction, withAddress: contractAddress };
+
+  const decimals = logEvent.sender_contract_decimals ?? undefined;
+  const symbol = logEvent.sender_contract_ticker_symbol || undefined;
+  const iconURL = logEvent.sender_logo_url ?? undefined;
+
+  if (logEvent.decoded.name === 'Transfer') {
+    const fromAddress = logEvent.decoded.params.at(0)!.value;
+    const toAddress = logEvent.decoded.params.at(1)!.value;
+
+    const type = (() => {
+      if (toAddress === accountAddress) {
+        return item.to_address === logEvent.sender_address
+          ? ActivityOperTransferType.receiveFromAccount
+          : ActivityOperTransferType.receive;
+      }
+
+      if (fromAddress === accountAddress) {
+        return item.to_address === logEvent.sender_address
+          ? ActivityOperTransferType.sendToAccount
+          : ActivityOperTransferType.send;
+      }
+
+      return null;
+    })();
+
+    if (type == null) return { kind: ActivityOperKindEnum.interaction, withAddress: contractAddress };
+
+    const kind = ActivityOperKindEnum.transfer;
+
+    if (!contractAddress) return { kind, type, fromAddress, toAddress };
+
+    const param3 = logEvent.decoded.params.at(2);
+    const amountOrTokenId: string = param3?.value ?? '0';
+    const nft = param3?.indexed ?? false;
+    const tokenId = nft ? amountOrTokenId : undefined;
+
+    const amount = nft ? '1' : amountOrTokenId;
+
+    const amountSigned =
+      type === ActivityOperTransferType.send || type === ActivityOperTransferType.sendToAccount ? `-${amount}` : amount;
+
+    const asset: EvmActivityAsset = {
+      contract: contractAddress,
+      tokenId,
+      amountSigned,
+      decimals,
+      symbol,
+      nft,
+      iconURL
+    };
+
+    return { kind, type, fromAddress, toAddress, asset };
+  }
+
+  if (logEvent.decoded.name === 'TransferSingle') {
+    const fromAddress = logEvent.decoded.params.at(1)!.value;
+    const toAddress = logEvent.decoded.params.at(2)!.value;
+
+    const type = (() => {
+      if (toAddress === accountAddress) {
+        return item.to_address === logEvent.sender_address
+          ? ActivityOperTransferType.receiveFromAccount
+          : ActivityOperTransferType.receive;
+      }
+
+      if (fromAddress === accountAddress) {
+        return item.to_address === logEvent.sender_address
+          ? ActivityOperTransferType.sendToAccount
+          : ActivityOperTransferType.send;
+      }
+
+      return null;
+    })();
```

Comment:
Please create a separate function for this logic; the same logic is here https://github.com/madfish-solutions/templewallet-extension/pull/1207/files#diff-c1a8250231f6e232bf3e9cd2f33a163d4fd1b5ba89aa4813119c4c8eecbba6f1R58